OrbitCore/TracepointEventBuffer.h

```diff
@@ -16,14 +16,22 @@
 
 class TracepointEventBuffer {
  public:
-
   void AddTracepointEventAndMapToThreads(uint64_t time, uint64_t tracepoint_hash,
                                          int32_t process_id, int32_t thread_id, int32_t cpu,
                                          bool is_same_pid_as_target);
 
   [[nodiscard]] const std::map<uint64_t, orbit_client_protos::TracepointEventInfo>&
   GetTracepointsOfThread(int32_t thread_id) const;
 
+  void ForEachTracepointEventOfThread(
+      int32_t thread_id, uint64_t min_tick, uint64_t max_tick,
+      const std::function<void(const orbit_client_protos::TracepointEventInfo&)>& action) const;
+
+  void ForEachTracepointEventOfThreadInTimeRange(
```

Comment:
Why do you need both methods? Can't you have a single public 
`ForEachTracepointEventOfThreadInTimeRange(int32_t thread_id, uint64_t min_tick, uint64_t max_tick,
      const std::function<void(const orbit_client_protos::TracepointEventInfo&)>& action) const` 
with the two nested for loops?

Reply:
Of course I could. I use 'ForEachTracepointEventOfThreadInTimeRange' in two places so I believed that not writing repeating code would be a good idea and would also make everything cleaner. Let me know if this is a good trade-off.

Reply:
Actually, since `ForEachTracepointEventOfThreadInTimeRange` is used twice and you might want to keep it, I would do like this:
`ForEachTracepointEventOfThread` should be the only public one and called `ForEachTracepointEventOfThreadInTimeRange`, as that's what it's actually doing;
`ForEachTracepointEventOfThreadInTimeRange` is an internal method: it should be private (and maybe renamed to avoid the overloading here).

Reply:
Actually the old `ForEachTracepointEventOfThreadInTimeRange` can just be a free-standing static function, and maybe rename to `ForEachTracepointEventInEventMapInTimeRange`

Reply:
Maybe I am missing something, but what overloading? I do not think I have another method with the same name but different parameters.

Reply:
To summarize:
old `ForEachTracepointEventOfThread` should become `ForEachTracepointEventOfThreadInTimeRange`;
old `ForEachTracepointEventOfThreadInTimeRange` should become a static free-standing function in the .cpp only, and be renamed to something like `ForEachTracepointEventInEventMapInTimeRange`.

Reply:
Thank you so much for the suggestions! I will update the code. 👍